src/test/ui/self/suggest-self-2.stderr

```diff
@@ -0,0 +1,34 @@
+error[E0425]: cannot find function `bar` in this scope
+  --> $DIR/suggest-self-2.rs:5:9
+   |
+LL |         bar(self);
+   |         ^^^ help: try calling `bar` as a method: `self.bar()`
```

Comment:
The span should be the entire call: `bar(self)`. Other than that, this looks good!

Reply:
Done :) Also changed the implementation a bit - hope for the better.